### ISSUE TITLE
[Modal] Make modal ARIA compliant with role="dialog" and fix keyboard accessibility issues.

### ISSUE THREAD

**Issue by Giwayume** [Semantic-Org/Semantic-UI#5053]
One huge thing this framework is missing is accessibility compliance. 

A few issues:
* role of dialog is missing
* aria-labelledby missing (assuming this could always be the header)
* first focusable control should be focused when modal is shown
* keyboard focus should be resumed on the last focused element before the modal was shown, after the modal is hidden
* tabbing order should wrap inside the modal and not allow focus to escape the modal while open

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role

**Comment by Giwayume** [comment]
The above was based on the current release. I see after downloading the following items have already been addressed:
* first focusable control should be focused when modal is shown
* keyboard focus should be resumed on the last focused element before the modal was shown, after the modal is hidden
